apps/interactor/interactor/tasks/registered/circadian/options.py

```diff
@@ -0,0 +1,160 @@
+from interactor.tasks.registered.circadian.circadian import Circadian
+from interactor.tasks.time_specs import time_spec, time_range_spec
+from interactor.tasks.registered.circadian.specs import days_spec
+from interactor.tasks.register import DeviceTask
+
+from photons_app.errors import PhotonsAppError
+from photons_app import helpers as hp
+
+from delfick_project.norms import dictobj, sb
+from astral.geocoder import database, lookup
+from astral.location import Location
+from astral import LocationInfo
+from datetime import datetime
+
+
+class NoSuchLocation(PhotonsAppError):
+    desc = "Couldn't find this location"
+
+
+class location_spec(sb.Spec):
+    def normalise_filled(self, meta, val):
+        if isinstance(val, str):
+            try:
+                info = Location(lookup(val, database()))
+            except KeyError:
+                raise NoSuchLocation(want=val, meta=meta)
+        else:
+            options = sb.set_options(
+                name=sb.required(sb.string_spec()),
+                region=sb.required(sb.string_spec()),
+                timezone=sb.required(sb.string_spec()),
+                latitude=sb.required(sb.integer_spec()),
```

Comment:
Latitude and longitude as integers means the closest location I can provide to my actual location is over 40km away in the middle of Port Philip Bay. Given that Google Maps provides latitude and longitude values with 14 decimal places, perhaps a float might be better?

Reply:
oops, fixed